server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/grepSearch.ts

```diff
@@ -0,0 +1,315 @@
+        try {
+            const results = await this.executeRipgrep(params, path, token)
+            return this.createOutput(results)
+        } catch (error: any) {
+            if (CancellationError.isUserCancelled(error)) {
+                // bubble this up to the main agentic chat loop
+                throw error
+            }
+            this.logging.error(`Failed to search in \"${path}\": ${error.message || error}`)
+            throw new Error(`Failed to search in \"${path}\": ${error.message || error}`)
+        }
+    }
+
+    private getRipgrepLibraryPath(): string {
+        if (process.platform === 'win32') {
+            // On Windows, the path must be loaded using a URL.
+            // Using the file path directly results in ERR_UNSUPPORTED_ESM_URL_SCHEME
+            // More details: https://github.com/nodejs/node/issues/31710
+            return pathToFileURL(path.join(RIPGREP_DIR, 'rg.exe')).toString()
+        }
+
+        return path.join(RIPGREP_DIR, 'rg')
+    }
+
+    private getSearchDirectory(path?: string): string {
+        if (path && path.trim().length !== 0) {
+            return path
+        }
+
+        // Use current workspace folder as default if path is not provided
+        const workspaceFolders = getWorkspaceFolderPaths(this.workspace)
+        if (workspaceFolders && workspaceFolders.length !== 0) {
+            this.logging.debug(`Using default workspace folder: ${workspaceFolders[0]}`)
+            return workspaceFolders[0]
+        }
+
+        return ''
+    }
+
+    private async executeRipgrep(
+        params: GrepSearchParams,
+        path: string,
+        token?: CancellationToken
+    ): Promise<SanitizedRipgrepOutput> {
+        return new Promise(async (resolve, reject) => {
+            const args: string[] = []
+
+            // Add search options
+            if (!(params.caseSensitive ?? false)) {
+                args.push('-i') // Case insensitive search
+            }
+            args.push('-n') // Show line numbers
+
+            // No heading (don't group matches by file)
+            args.push('--no-heading')
+
+            // Don't use color in output
+            args.push('--color=never')
+
+            // Limit results to prevent overwhelming output
+            args.push('-m', '20')
+
+            // Add include/exclude patterns
+            if (params.includePattern) {
+                // Support multiple include patterns
+                const patterns = params.includePattern.split(',')
+                for (const pattern of patterns) {
+                    args.push('-g', `${pattern.trim()}`)
+                }
+            }
+
+            if (params.excludePattern) {
+                // Support multiple exclude patterns
+                const patterns = params.excludePattern.split(',')
+                for (const pattern of patterns) {
+                    args.push(`-g`, `!${pattern.trim()}`)
+                }
+            }
```

Comment:
we should also update the tool spec and implementation to ignore the common build and dependency directories.

Reply:
we don't ignore them

Reply:
ripgrep respects gitignore rules already and there is no problem searching for more files because it's so fast